apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -92,139 +108,108 @@ public static System.Type GetDefaultClrTypeFromSpannerType(this TypeCode code)
-                    throw new ArgumentOutOfRangeException(nameof(typeCode), typeCode, null);
-            }
-        }
-
-        public static TypeCode GetSpannerType(this DbType dbType)
+        public static TypeCode GetSpannerTypeCode(this SpannerDbType spannerDbType)
         {
-            switch (dbType)
-            {
-                case DbType.Binary:
-                    return TypeCode.Bytes;
-                case DbType.Boolean:
-                    return TypeCode.Bool;
-                case DbType.Date:
-                    return TypeCode.Date;
-                case DbType.DateTime:
-                    return TypeCode.Timestamp;
-                case DbType.Double:
-                    return TypeCode.Float64;
-                case DbType.Int64:
-                    return TypeCode.Int64;
-                case DbType.Object:
-                    return TypeCode.Unspecified;
-                case DbType.String:
-                    return TypeCode.String;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(dbType), dbType, null);
-            }
+            return spannerDbType.TypeCode;
         }
 
-        public static TypeCode GetSpannerTypeCode(this SpannerDbType spannerDbType)
+        public static Value ToValue(object value, SpannerDbType spannerDbType)
         {
-            switch (spannerDbType)
+            if (value == null)
             {
-                case SpannerDbType.Unspecified:
-                    return TypeCode.Unspecified;
-                case SpannerDbType.Bool:
-                    return TypeCode.Bool;
-                case SpannerDbType.Int64:
-                    return TypeCode.Int64;
-                case SpannerDbType.Float64:
-                    return TypeCode.Float64;
-                case SpannerDbType.Timestamp:
-                    return TypeCode.Timestamp;
-                case SpannerDbType.Date:
-                    return TypeCode.Date;
-                case SpannerDbType.String:
-                    return TypeCode.String;
-                case SpannerDbType.Bytes:
-                    return TypeCode.Bytes;
-                case SpannerDbType.Array:
-                    return TypeCode.Array;
-                case SpannerDbType.Struct:
-                    return TypeCode.Struct;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(spannerDbType), spannerDbType, null);
+                return Value.ForNull();
             }
-        }
 
-        public static Value ToValue(object value, TypeCode typeCode)
-        {
-            switch (typeCode)
+            switch (spannerDbType.TypeCode)
             {
+                case TypeCode.Bytes:
+                    var s = value as string;
+                    if (s != null)
+                    {
+                        return new Value { StringValue = s };
+                    }
+                    else if (value is byte[])
```

Comment:
Remove the `else` and make this an `as` casting as well so you don't need to double-cast.